src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1015,6 +1015,12 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
                 {
                     HttpResponseMessage? response = null;
 
+                    // We could not use version higher than HTTP/3. Do not continue if downgrade is not allowed.
+                    if (request.Version.Major > 3 && request.VersionPolicy != HttpVersionPolicy.RequestVersionOrLower)
+                    {
+                        ThrowGetVersionException(request, request.Version.Major);
+                    }
```

Comment:
Is this the right fix, or should this condition be changed:
https://github.com/dotnet/runtime/pull/73461/files#diff-bc2faad6c76ecb9b115e305763eb55ebfde74de1eee27ed549e5edaabb097a99R1028
?

Also, if this does remain, it seems weird that, in your example, the resulting error text will contain "HTTP/4".